packages/server/src/integrations/googlesheets.ts

```diff
@@ -345,18 +349,35 @@ class GoogleSheetsIntegration implements DatasourcePlus {
     }
   }
 
-  async read(query: { sheet: string }) {
+  async read(query: {
+    sheet: string
+    filters?: SearchFilters
+    sort?: SortJson
+    paginate?: PaginationJson
+  }) {
     try {
       await this.connect()
       const sheet = this.client.sheetsByTitle[query.sheet]
       const rows = await sheet.getRows()
+      const filtered = dataFilters.runLuceneQuery(rows, query.filters)
       const headerValues = sheet.headerValues
-      const response = []
-      for (let row of rows) {
+      let response = []
+      for (let row of filtered) {
         response.push(
           this.buildRowObject(headerValues, row._rawData, row._rowNumber)
         )
       }
+
+      if (query.sort) {
+        const sortInfo = Object.entries(query.sort)[0]
+        response = dataFilters.luceneSort(
+          response,
+          sortInfo[0],
```

Comment:
NAB: Is there a way to make this a little bit more readable / use types instead of `sortInfo[0]` and `sortInfo[1]` ?

Reply:
Cleaned